contracts/protocol/modules/LimitOrder.sol

```diff
@@ -162,30 +162,32 @@ abstract contract LimitOrderManager is ClaimableAmountManager {
                 if (
                     _rawQuantity - closePositionResp.exchangedPositionSize == 0
                 ) {
-                    // TODO deposit margin to vault of position resp
-                    //                            positionResp = closePositionResp;
-                    //                            deposit(_positionManager, _trader, positionResp.marginToVault.abs(), 0);
+                    sizeOut = _rawQuantity.abs();
+                    if (closePositionResp.marginToVault < 0) {
+                        withdraw(_positionManager, _trader, closePositionResp.marginToVault.abs());
+                    }
                 } else {
                     _quantity -= (closePositionResp.exchangedPositionSize)
                         .abs128();
                 }
-            }
-            (orderId, sizeOut, openNotional) = _positionManager
-                .openLimitPosition(_pip, _quantity, _rawQuantity > 0);
-            if (sizeOut != 0) {
-                // case: open a limit order at the last price
-                // the order must be partially executed
-                // then update the current position
-                Position.Data memory newData;
-                newData = PositionHouseFunction.handleMarketPart(
-                    oldPosition,
-                    _getPositionMap(_pmAddress, _trader),
-                    openNotional,
-                    _rawQuantity > 0 ? int256(sizeOut) : -int256(sizeOut),
-                    _leverage,
-                    getCumulativePremiumFractions(_pmAddress)
-                );
-                _updatePositionMap(_pmAddress, _trader, newData);
+            } else {
```

Comment:
just one case can happen: close old position or reduce position and create a new limit order